backends/ebpf/psa/ebpfPsaTable.cpp

```diff
@@ -86,6 +87,43 @@ EBPFTablePSA::EBPFTablePSA(const EBPFProgram* program, const IR::TableBlock* tab
         }
         this->size = 1;
     }
+
+    initImplementation();
+}
+
+EBPFTablePSA::EBPFTablePSA(const EBPFProgram* program, CodeGenInspector* codeGen, cstring name) :
+                           EBPFTable(program, codeGen, name), implementation(nullptr) {}
+
+void EBPFTablePSA::initImplementation() {
+    // PSA table is allowed to have up to one table implementation. EBPFTablePsaPropertyVisitor
+    // will iterate over all entries in property, so lets use this and print errors.
+    auto impl = [this](const IR::PathExpression * pe) {
```

Comment:
I am not sure why you chose to split the implementation into a lambda and a visitor, this code could have been in the preorder method for PathExpression.
BTW: if that class is only used here you can move it to the cpp file.

Reply:
DirectCounter and  DirectMeter are added to a table in the same way but other lambdas are used, this the reason why I split implementation.